offchain/state-server/src/lib.rs

```diff
@@ -8,19 +21,36 @@ use state_server_lib::{
     utils::{start_server, wait_for_signal},
 };
 
-use anyhow::Result;
+use snafu::{ResultExt, Snafu};
 use std::sync::{Arc, Mutex};
 use tokio::sync::oneshot;
 use types::UserData;
-use url::Url;
+use url::{ParseError, Url};
 
 const MAX_RETRIES: u32 = 10;
 const INITIAL_BACKOFF: u64 = 1000;
 
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)))]
+pub enum StateServerError {
```

Comment:
Shouldn't this struct have its own file as you did for the other services?

Reply:
It can be, for sure, but since the state-server is such a small crate compared to the others, I thought it would bloat it to add another file just for the errors. I can separate it, no problem.

Reply:
The inner battle of keeping things small vs. keeping standards is strong here.
But I think I agree with Gabriel.